backend/src/BuiltinExecution/Libs/Float.fs

```diff
@@ -269,20 +281,31 @@ let fns : List<BuiltInFn> =
     { name = fn "parse" 0
       typeParams = []
       parameters = [ Param.make "s" TString "" ]
-      returnType = TypeReference.result TFloat TString
+      returnType =
+        TypeReference.result
+          TFloat
+          (TCustomType(
+            Ok(
+              FQName.Package
+                { owner = "Darklang"
+                  modules = [ "Stdlib"; "Int" ]
+                  name = TypeName.TypeName "IntParseError"
+                  version = 0 }
+            ),
+            []
+          ))
       description =
         "Returns the <type Float> value wrapped in a {{Result}} of the <type String>"
       fn =
         let resultOk r = Dval.resultOk KTFloat KTString r |> Ply
-        let resultError r = Dval.resultError KTFloat KTString r |> Ply
+        let typeName = RuntimeError.name [ "Float" ] "FloatParseError" 0
+        let resultError = Dval.resultError KTFloat (KTCustomType(typeName, []))
         (function
         | _, _, [ DString s ] ->
           try
             float (s) |> DFloat |> resultOk
-          with e ->
-            "Expected a String representation of an IEEE float"
-            |> DString
-            |> resultError
+          with :? System.FormatException ->
```

Comment:
I don't think we need to catch the `OverflowException`
here since the result is rounded to infinity, wdyt?

> In .NET Core 3.0 and later, values that are too large to represent are rounded to [PositiveInfinity] or [NegativeInfinity] as required by the IEEE 754 specification. In prior versions, including .NET Framework, parsing a value that was too large to represent resulted in failure.

https://learn.microsoft.com/en-us/dotnet/api/system.double.parse?view=net-7.0